src/main/java/edu/usf/cutr/grha/otp/OtpService.kt

```diff
@@ -0,0 +1,105 @@
+package edu.usf.cutr.grha.otp
+
+import edu.usf.cutr.grha.model.ChicagoTncData
+import edu.usf.cutr.grha.utils.GtfsUtils
+import edu.usf.cutr.otp.plan.api.PlanApi
+import edu.usf.cutr.otp.plan.model.Planner
+import edu.usf.cutr.otp.plan.model.RequestParameters
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.SingleOnSubscribe
+import kotlinx.coroutines.*
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.flow.*
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.resumeWithException
+
+class OtpService(
+    private var chicagoTncData: List<ChicagoTncData>,
+    private var url: String
+) {
+
+    fun call() {
+        makePlanFlow()
+    }
+
+    private fun getPlanData() {
+        var i = 0
+        var start = 0L
+        val planFlow: Flow<Planner> = flow {
+            while (i < 42) {
+                if (chicagoTncData[i].pickupCentroidLatitude == 0.0 ||
+                    chicagoTncData[i].pickupCentroidLongitude == 0.0 ||
+                    chicagoTncData[i].dropoffCentroidLongitude == 0.0 ||
+                    chicagoTncData[i].dropoffCentroidLongitude == 0.0
+                ) {
+                    i++
+                    continue
+                }
+                val origin = GtfsUtils.latLong(
+                    chicagoTncData[i].pickupCentroidLatitude,
+                    chicagoTncData[i].pickupCentroidLongitude
+                )
+                val destination = GtfsUtils.latLong(
+                    chicagoTncData[i].dropoffCentroidLongitude,
+                    chicagoTncData[i].dropoffCentroidLongitude
+                )
+                val requestParameters = RequestParameters(fromPlace = origin, toPlace = destination)
+                start = System.currentTimeMillis()
+                val planApi = PlanApi(url, requestParameters)
+                emit(makePlanRequest(planApi, i))
+                i++
+            }
+        }
+        runBlocking {
+            planFlow.collect {
+                val time = System.currentTimeMillis() - start
+                println(it)
+                println("Time: $time ms")
+            }
+        }
+    }
+
+    private fun makePlanFlow() {
+        val dataFlow = chicagoTncData.asFlow()
+        runBlocking {
+            dataFlow
+                .filter {
+                    it.pickupCentroidLatitude != 0.0
```

Comment:
I see now you'd only using this method because it's faster 🚀 - same on moving check to private method.